🪧 feat: Add Custom Group Icon Support for Model Specs

Odrec opened the pull request:
Added the ability to define icons for custom model spec groups in the UI selector.

Motivation: When using modelSpecs with custom group names (not matching endpoint names), the groups appear in the dropdown without any visual icon, making the list look inconsistent. This feature allows users to define a groupIcon property to display an icon next to the group name, improving the visual appearance of the model selector.

**Changes:**

- Added groupIcon property to TModelSpec type and schema in packages/data-provider/src/models.ts
- Created GroupIcon component to render URL or built-in endpoint icons
- Updated CustomGroup component to display group icons
- Added documentation and examples in librechat.example.yaml

**Usage:**

```
modelSpecs:
  list:
    - name: "my-model"
      group: "Custom Group"
      groupIcon: "openAI"  # or a URL
      preset:
        endpoint: "openAI"
        model: "gpt-4o"
```

The groupIcon can be:

- A built-in endpoint key (e.g., "openAI", "anthropic", "groq")
- A URL to a custom icon image

Only the first spec in a group needs groupIcon - all specs in that group share the same icon.

**Change Type**
 - [✓] New feature (non-breaking change which adds functionality)
 - [✓] This change requires a documentation update

**Testing**

1. Add modelSpecs configuration to librechat.yaml with custom groups and groupIcon properties
Start the backend and frontend development servers
2. Log in and open the model selector dropdown
3. Verify that custom groups display the specified icons next to their names


**Checklist**
 - [✓] My code adheres to this project's style guidelines
 - [✓] I have performed a self-review of my own code
 - [✓] I have commented in any complex areas of my code
 - [] I have made pertinent documentation changes
 - [✓] My changes do not introduce new warnings
 - [] I have written tests demonstrating that my changes are effective or that my feature works
 - [✓] Local unit tests pass with my changes
 - [] A pull request for updating the documentation has been submitted.

Odrec committed:
feat: add groupIcon property to modelSpecs for custom group icons

Added the ability to define icons for custom model spec groups in the UI selector.

Changes:
- Added  property to TModelSpec type and schema in data-provider
- Created GroupIcon component to render URL or built-in endpoint icons
- Updated CustomGroup component to display group icons
- Added documentation and examples in librechat.example.yaml

Usage:

The groupIcon can be:
- A built-in endpoint key (e.g., "openAI", "anthropic", "groq")
- A URL to a custom icon image

Only the first spec in a group needs groupIcon - all specs share the same icon.

Odrec committed:
Merge branch 'main' into feat/model-spec-group-icons

Odrec committed:
Merge branch 'main' into feat/model-spec-group-icons

Odrec committed:
Merge branch 'main' into feat/model-spec-group-icons

Odrec committed:
Merge branch 'main' into feat/model-spec-group-icons

usnavy13 commented:
@Odrec you might want to also change this pr from the main branch to the dev branch

Odrec commented:
> @Odrec you might want to also change this pr from the main branch to the dev branch

thx for the heads up!

Odrec committed:
Merge branch 'dev' into feat/model-spec-group-icons

danny-avila commented:
@Odrec can you address the Copilot comments? I've reviewed them and they appear to be valid

Odrec commented:
> @Odrec can you address the Copilot comments? I've reviewed them and they appear to be valid

Thx! I'll have time to look into this next week.

Commit:
Merge remote-tracking branch 'upstream/dev' into feat/model-spec-group-icons

Commit:
fix: address Copilot review comments for GroupIcon component

- Changed URL detection from includes('http') to checking if iconURL exists in icons map (more robust approach)
- Removed redundant !iconURL check since iconURL is always a string from props

Odrec commented:
@danny-avila The changes were made following the copilot review.

danny-avila commented:
Merging soon, adding documentation would be greatly appreciated if anyone can get to it:

https://github.com/LibreChat-AI/librechat.ai/pulls

@dustinhealy @Odrec
